lib/pure/asynchttpserver.nim

```diff
@@ -52,6 +55,7 @@ type
     url*: Uri
     hostname*: string    ## The hostname of the client that made the request.
     body*: string
+    bodyStream*: FutureStream[string]
```

Comment:
```nim

when (NimMajor, NimMinor) >= (1, 1):
  bodyStream*: FutureStream[string]
```

Reply:
> ```nim
> when (NimMajor, NimMinor) >= (1, 1):
>   bodyStream*: FutureStream[string]
> ```
When I put the annotation I get this error:
Error: constant expression expected
```nim
type
  Request* = object
    client*: AsyncSocket # TODO: Separate this into a Response object?
    reqMethod*: HttpMethod
    headers*: HttpHeaders
    protocol*: tuple[orig: string, major, minor: int]
    url*: Uri
    hostname*: string    ## The hostname of the client that made the request.
    body*: string
    when (NimMajor, NimMinor) >= (1, 1):
      bodyStream*: FutureStream[string]
```

Reply:
Bah, what's up with that...

Reply:
Works if outside of type declaration
```nim
when (NimMajor, NimMinor) >= (1, 1):
  const
    maxLine = 8*1024
    chunkSize = 1048

  type
    Request* = object
      client*: AsyncSocket # TODO: Separate this into a Response object?
      reqMethod*: HttpMethod
      headers*: HttpHeaders
      protocol*: tuple[orig: string, major, minor: int]
      url*: Uri
      hostname*: string    ## The hostname of the client that made the request.
      body*: string
      bodyStream*: FutureStream[string] # <<<<<
else:
  const
    maxLine = 8*1024

  type
    Request* = object
      client*: AsyncSocket # TODO: Separate this into a Response object?
      reqMethod*: HttpMethod
      headers*: HttpHeaders
      protocol*: tuple[orig: string, major, minor: int]
      url*: Uri
      hostname*: string    ## The hostname of the client that made the request.
      body*: string

type
  AsyncHttpServer* = ref object
    socket: AsyncSocket
    reuseAddr: bool
    reusePort: bool
    maxBody: int ## The maximum content-length that will be read for the body.
```